pkg/logql/engine.go

```diff
@@ -139,6 +139,10 @@ type Querier interface {
 	SelectSamples(context.Context, SelectSampleParams) (iter.SampleIterator, error)
 }
 
+type Engine interface {
+	Query(Params) Query
+}
```

Comment:
aside:

Typically I think this would be defined where the two different engine implementations would be consumed rather than where one of the implementations is provided (it would avoid needing to rename one of the types).

But that's really if you're following by-the-book Go conventions; I rarely follow this rule myself so I think it's perfectly fine to keep it this way.

Reply:
Agree. However, it did not feel quite right to put the interface into the `querier` package.
Also, the compile-type interface assertion would need to happen in the querier package, otherwise there would be a circular dependency.